Tests/Services/HealthTests/HealthTests.swift

```diff
@@ -0,0 +1,322 @@
+        let signal = AsyncStream.makeStream(of: Void.self)
+        let numberOfWatches = 2
+
+        for _ in 0 ..< numberOfWatches {
+          group.addTask {
+            return try await healthClient.watch(
+              request: ClientRequest.Single(message: immutableMessage)
+            ) { response in
+              signal.continuation.yield()  // Make signal
+
+              var statuses = [Grpc_Health_V1_HealthCheckResponse.ServingStatus]()
+              var responseStreamIterator = response.messages.makeAsyncIterator()
+
+              // Since responseStreamIterator.next() will never be nil (ideally, as the response
+              // stream is always open), the iteration cannot be based on when
+              // responseStreamIterator.next() is nil. Else, the iteration infinitely awaits and the
+              // test never finishes. Hence, it is based on the expected number of statuses to be
+              // received.
+              for _ in 0 ..< statusesToBeSent.count + 1 {
+                // As the service will be "watched" before being updated, the first status received
+                // should be `.serviceUnknown`. Hence, the range of this iteration is increased by 1.
+
+                let next = try await responseStreamIterator.next()
+                let message = try XCTUnwrap(next)
+                statuses.append(message.status)
+              }
+
+              return statuses
+            }
+          }
+        }
+
+        // Wait until all the "watch" streams are up and ready.
+        for await _ in signal.stream.prefix(numberOfWatches) {}
+
+        for status in statusesToBeSent {
+          healthProvider.updateStatus(status, ofService: testServiceDescriptor)
+        }
+
+        for try await receivedStatuses in group {
+          XCTAssertEqual(receivedStatuses[0], .serviceUnknown)
+
+          for i in 0 ..< statusesToBeSent.count {
+            let sentStatus = Grpc_Health_V1_HealthCheckResponse.ServingStatus(statusesToBeSent[i])
+            XCTAssertEqual(sentStatus, receivedStatuses[i + 1])
+          }
+        }
+      }
+    }
+  }
+
+  func testWatchWithUnchangingStatusUpdates() async throws {
+    try await withHealthClient { (healthClient, healthProvider) in
+      let testServiceDescriptor = ServiceDescriptor.testService
+
+      let statusesToBeSent: [ServingStatus] = [
+        .serving,
+        .notServing,
+        .notServing,
+        .notServing,
+        .serving,
+      ]
+
+      let expectedReceivedStatuses: [Grpc_Health_V1_HealthCheckResponse.ServingStatus] = [
+        // As the service will be "watched" before being updated, the first status received should
+        // be .serviceUnknown.
+        .serviceUnknown,
+        .serving,
+        .notServing,  // The repeated `.notServing` updates should be received only once.
+        .serving,
+      ]
+
+      try await withThrowingTaskGroup(
```

Comment:
I don't think this test needs a task group, we should be able to do everything we need within `watch`:

```swift
healthClient.watch(...) { response in 
  var iterator = response.messages.makeAsyncIterator()
  // wait for 'serviceUnknown'
  // send status updates
  // read updates
}
```